### PR TITLE
Relation metadata

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] Feature
- [x] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

The information of which sides a relation is connected is encoded in `Connector` objetcts

Issue Number: #648

### What is the new behavior?

Relation (Presentation) elements provide extra metadata about which ends a line will connect. This information can then be reused when dropping a relation from the tree view on a diagram.

### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### Other information

Builds on #1418.

### CODE REVIEW

File: gaphor/diagram/connectors.py [hunk 1]
```diff
@@ -364,6 +363,45 @@ def allow(self, handle: Handle, port: Port) -> bool:
...
+                or isinstance(opposite_element.subject, metadata["tail"].type)
+            )
+        else:
+            return isinstance(element.subject, metadata["tail"].type) and (
+                not opposite_element
+                or isinstance(opposite_element.subject, metadata["head"].type)
+            )
+
+    def connect_subject(self, handle):
+        print("MetadataRelationConnect", self.element, self.line)
```
Comment (danyeaw) [inline gaphor/diagram/connectors.py:393]: Can we remove this print statement?

File: gaphor/UML/classes/classconnect.py [hunk 2]
```diff
@@ -169,34 +157,3 @@ def disconnect_subject(self, handle) -> None:
         del self.line.head_subject
         del self.line.tail_subject
         super().disconnect_subject(handle)
-
-
-@Connector.register(Named, InterfaceRealizationItem)
-class InterfaceRealizationConnect(DirectionalRelationshipConnect):
```
Comment (danyeaw) [inline gaphor/UML/classes/classconnect.py:175]: I really like that you were able to remove all of these extra Connect classes!

**Review by danyeaw — APPROVED** [review]
Looks great!

**Comment by amolenaar** [comment]
I cleaned up connectors for sysml and raaml as well. Added a test that shows what's covered and what's not.

**Pull request merged**